### PR TITLE
Feature: Get all users endpoint

### PR DESCRIPTION
**Pull request by Miles-Alford**
### *What is the problem we're trying to solve*

This PR will solve the issue of allowing the front-end to receive all users stored within the database. For now this will primarily be used for adding a user to a project by providing the values for the autocomplete input on the front-end.

### *What changes have we introduced*
Add the complete endpoint for getting all users stored within the database

### *Follow up actions after merging PR*
Depending on best practice either this functionality or the future endpoint for getting a user by their email will be used for a validation check on the POST endpoint

### CODE REVIEW

File: ProjectFinderApi/Dockerfile [hunk 1]
```diff
@@ -11,5 +11,5 @@ COPY . ./
 RUN dotnet build -c Release -o out
 RUN dotnet publish -c Release -o out
 
-EXPOSE ${PORT:-3000}
-CMD ASPNETCORE_URLS=http://+:${PORT:-3000} dotnet ./out/ProjectFinderApi.dll
+EXPOSE ${PORT:-8080}
+CMD ASPNETCORE_URLS=http://+:${PORT:-8080} dotnet ./out/ProjectFinderApi.dll
```
Comment (LBHTKarki) [inline ProjectFinderApi/Dockerfile:15]: this port is commonly used by many default setups running locally, so might be worth picking a different one to avoid clashes later.

File: ProjectFinderApi/V1/Controllers/UserController.cs [hunk 2]
```diff
@@ -46,5 +47,22 @@ public IActionResult CreateUser([FromBody] CreateUserRequest request)
                 return UnprocessableEntity(e.Message);
             }
         }
+
+        /// <summary>
+        /// Get all users
+        /// </summary>
+        /// <response code="404">No users found.</response>
```
Comment (LBHTKarki) [inline ProjectFinderApi/V1/Controllers/UserController.cs:54]: Would be good to add all responses that are possibly returned by the controller, such as 500 and 200

File: ProjectFinderApi/V1/Gateways/UsersGateway.cs [hunk 3]
```diff
@@ -29,5 +30,10 @@ public User CreateUser(CreateUserRequest createUserRequest)
             _databaseContext.SaveChanges();
             return user;
         }
+
+        public IEnumerable<User> GetUsers()
```
Comment (LBHTKarki) [inline ProjectFinderApi/V1/Gateways/UsersGateway.cs:34]: I know you'll be adding paging in a separate PR, so this can stay as it is for now.

**Review by LBHTKarki — APPROVED** [review]

File: ProjectFinderApi/Dockerfile (hunk 1 above)
Comment (Miles-Alford) [reply ProjectFinderApi/Dockerfile]: 👍
